crates/verify/src/sourcify.rs

```diff
@@ -47,155 +64,317 @@ impl VerificationProvider for SourcifyVerificationProvider {
         context: &VerificationContext,
     ) -> Result<SourcifyVerifyRequest> {
-        let metadata = context.get_target_metadata()?;
-        let imports = context.get_target_imports()?;
+        let lang = args.detect_language(context);
 
-        let mut files = HashMap::with_capacity_and_hasher(2 + imports.len(), Default::default());
+        let std_json_input = match lang {
+            ContractLanguage::Solidity => {
+                let mut input: StandardJsonCompilerInput = context
+                    .project
+                    .standard_json_input(&context.target_path)
+                    .wrap_err("Failed to get standard json input")?
+                    .normalize_evm_version(&context.compiler_version);
 
-        let metadata = serde_json::to_string_pretty(&metadata)?;
-        files.insert("metadata.json".to_string(), metadata);
+                let mut settings = context.compiler_settings.solc.settings.clone();
+                settings.libraries.libs = input
+                    .settings
+                    .libraries
+                    .libs
+                    .into_iter()
+                    .map(|(f, libs)| {
+                        (f.strip_prefix(context.project.root()).unwrap_or(&f).to_path_buf(), libs)
+                    })
+                    .collect();
 
-        let contract_path = context.target_path.clone();
-        let filename = contract_path.file_name().unwrap().to_string_lossy().to_string();
-        files.insert(filename, fs::read_to_string(&contract_path)?);
+                settings.remappings = input.settings.remappings;
 
-        for import in imports {
-            let import_entry = format!("{}", import.display());
-            files.insert(import_entry, fs::read_to_string(&import)?);
-        }
+                // remove all incompatible settings
+                settings.sanitize(&context.compiler_version, SolcLanguage::Solidity);
+
+                input.settings = settings;
+
+                serde_json::to_value(&input).wrap_err("Failed to serialize standard json input")?
+            }
+            ContractLanguage::Vyper => {
+                let path = Path::new(&context.target_path);
+                let sources = Source::read_all_from(path, &["vy", "vyi"])?;
+                let input = VyperInput::new(
+                    sources,
+                    context.clone().compiler_settings.vyper,
+                    &context.compiler_version,
+                );
+
+                serde_json::to_value(&input).wrap_err("Failed to serialize vyper json input")?
+            }
+        };
+
+        let contract_identifier = format!(
+            "{}:{}",
+            context
+                .target_path
+                .strip_prefix(context.project.root())
+                .unwrap_or(context.target_path.as_path())
+                .display(),
+            context.target_name
+        );
+
+        let compiler_version = if matches!(lang, ContractLanguage::Vyper) {
```

Comment:
can we move this in match above and return both `std_json_input` and `compiler_version` or maybe the `Ok(SourcifyVerifyRequest)` directly